tests/nested/manual/core20-install-mode-shutdown-via-hook/task.yaml

```diff
@@ -0,0 +1,91 @@
+summary: Verify snapctl poweroff install-device hook usage
+
+systems: [ubuntu-20.04-64]
+
+environment:
+    NESTED_IMAGE_ID: core20-install-device
+    NESTED_BUILD_SNAPD_FROM_CURRENT: true
+    NESTED_ENABLE_TPM: true
+    NESTED_ENABLE_SECURE_BOOT: true
+
+details: |
+    This test checks support for shutting down the device at the end of install
+    mode via the install-device hook
+
+prepare: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    # needed for inspecting the partitions
+    snap install jq
+
+    KEY_NAME=$(nested_get_snakeoil_key)
+    SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+    SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+    echo "Grab and prepare the gadget snap"
+    snap download --basename=pc --channel="20/edge" pc
+    unsquashfs -d pc-gadget pc.snap
+
+    echo "Sign the shim binary"
+    nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+
+    echo "Add the install-device hook"
+    mkdir -p pc-gadget/meta/hooks
+    cp install-device pc-gadget/meta/hooks/install-device
+    snap pack pc-gadget/ extra-snaps/
+
+    "$TESTSTOOLS"/nested-state build-image core
+
+execute: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    # we expect the VM to shut itself down so don't wait for SSH, etc. to become
+    # available
+    echo "Start the VM, which should automatically shut itself down"
+
+    # below is extracted from nested_start_core_vm, we don't want the 
+    # nested_prepare_ssh to run yet, since it will fail when the VM shuts itself
+    # down, we need to manually run it later
+    CURRENT_NAME="$(nested_get_current_image_name)"
+    CURRENT_IMAGE="$NESTED_IMAGES_DIR/$CURRENT_NAME"
+    IMAGE_NAME="$(nested_get_image_name core)"
+    IMAGE_FILE="$NESTED_IMAGES_DIR/$IMAGE_NAME"
+    cp -v "$IMAGE_FILE" "$CURRENT_IMAGE"
+
+    NESTED_EXPECT_SHUTDOWN=1 nested_start_core_vm_unit "$IMAGE_FILE"
+
+    echo "Wait for the VM to become inactive as it installs and shuts itself down"
+    retry -n 200 --wait 1 sh -c '! systemctl is-active nested-vm'
```

Comment:
We use this on other places 
wait_for_service "$NESTED_VM" active

Reply:
well we need to check for inactive, but I used this helper, thanks for the suggestion